### ISSUE TITLE
model requirements that generate the probability distribution

### ISSUE THREAD

**Issue by sandeepnmenon** [cgnorthcutt/cleanlab#66]
I understand from the paper that the CL framework is model-agnostic.
However, is there a baseline average test/validation accuracy required for the model while doing the cross-validation.

Also, to clarify, Is the probability values to be used in cleanlab the probability score of the sample when that sample is considered in the hold-out set of k-cross-validation.

**Comment by cgnorthcutt** [comment]
Hi @sandeepnmenon 

Your question can be slightly altered. There are two sources of noise, the labels, and the model.

If you have perfect labels, how well should the model perform? As high accuracy as possible, (depending on the problem, accuracy might not be the right metric). For most computer vision task for example, accuracy of the model on perfect labels should be better than 60 percent. But cleanlab can still work with lower, it just won't work well, because your model doesn't work well. The better the model, the better cleanlab will work.

Now noise in the labels will also lower your hold out accuracy. But if the model is near perfect on correct labels, then the amount of noise should be proportional to the decrease in accuracy. In other words, for 60 percent noise, you should be able to get at least 40 percent accuracy. Cleanlab can still work well in these cases, as long as the model works well on correct labels. 

Take a look at the benchmarks in the confident learning paper for various amounts of noise.

In general, you want your accuracy to be as high as possible... how accurate depends on the class of problems and how good your model is. 


P. S. Yes, your understanding of cross validation and how it's used in cleanlab looks right, but note that you don't have to use cross validation.. you can also use a pretrained model as long as it was trained on a different dataset.

What matters is the predicted probabilities are for "held out data". It doesn't matter if you use cross validation or a model trained on an entirely different dataset.
